src/Timeline/TimelineFactory.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace AppBundle\Timeline;
+
+use AppBundle\Content\MediaFactory;
+use AppBundle\Entity\Timeline\Measure;
+use AppBundle\Entity\Timeline\Profile;
+use AppBundle\Entity\Timeline\Theme;
+use AppBundle\Entity\Media;
+use AppBundle\Entity\Timeline\ThemeMeasure;
+use AppBundle\Repository\MediaRepository;
+use Cocur\Slugify\Slugify;
+use League\Flysystem\Filesystem as Storage;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\File\File;
+
+class TimelineFactory
+{
+    /**
+     * @var Filesystem|null
+     */
+    private $filesystem;
+
+    /**
+     * @var MediaFactory
+     */
+    private $mediaFactory;
+
+    /**
+     * @var MediaRepository
+     */
+    private $mediaRepository;
+
+    /**
+     * @var Slugify
+     */
+    private $slugifier;
+
+    /**
+     * @var Storage
+     */
+    private $storage;
+
+    public function __construct(
+        MediaFactory $mediaFactory,
+        MediaRepository $mediaRepository,
+        Slugify $slugifier,
+        Storage $storage
+    ) {
+        $this->mediaFactory = $mediaFactory;
+        $this->mediaRepository = $mediaRepository;
+        $this->slugifier = $slugifier;
+        $this->storage = $storage;
+
+    }
+
+    public function createProfile(string $title, string $description): Profile
+    {
+        $profile = new Profile();
+
+        $profile->setTitle($title);
+        $profile->setSlug($this->slugify($title));
+        $profile->setDescription($description);
```

Comment:
Could deserve a static constructor instead of repeated setter calls (same below).